src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/NuGetProjectManagerService.cs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft;
+using Microsoft.ServiceHub.Framework;
+using Microsoft.ServiceHub.Framework.Services;
+using NuGet.Packaging;
+using NuGet.ProjectManagement;
+using NuGet.VisualStudio;
+using NuGet.VisualStudio.Internal.Contracts;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    public sealed class NuGetProjectManagerService : INuGetProjectManagerService
+    {
+        private readonly ServiceActivationOptions _options;
+        private readonly IServiceBroker _serviceBroker;
+        private readonly AuthorizationServiceClient _authorizationServiceClient;
+
+        public NuGetProjectManagerService(ServiceActivationOptions options, IServiceBroker sb, AuthorizationServiceClient ac)
+        {
+            _options = options;
+            _serviceBroker = sb;
+            _authorizationServiceClient = ac;
+        }
+
+        public async ValueTask<IReadOnlyCollection<PackageReference>> GetInstalledPackagesAsync(IReadOnlyCollection<string> projectGuids, CancellationToken ct)
```

Comment:
There's a couple of places in the PM UI where we use BuildIntegratedPackageReference, so not sure if the PackageReference types is good enough here: https://github.com/NuGet/NuGet.Client/blob/40dcaa13be94b052b47227e08be36e2349c7656e/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Common/PackageCollectionItemExtensions.cs#L14

We might need a better type here.

Reply:
PackageReference is coming from NuGet.Packaging so changing that is probably out of the question, but having a model that's a bit more powerful on the VS end would be an acceptable change.  (note that change would be scope creep, but I don't think we should move on with this PR before addressing it). 

https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Packaging/PackageReference.cs

Reply:
I am not so sure this would be something that we should be concerned with.  The only thing calling the new service to get the PackageReferences would be the VisualStudioNuGetProject which is only used in the VSOe scenario and only client side, I would imagine that most of the processing of this type would be done server side which would be using the actual NuGetProject with the actual BuildIntegratedPackageReference object instead of the basic serialized/remoted one

Reply:
PackageReference doesn't contain all the data that BuildIntegratedPackageReference does. 

Additionally @martinrrm is going to be adding more details in the PM UI for PackageReference projects, https://github.com/NuGet/Home/issues/9827, as part of https://github.com/NuGet/Home/issues/3788. 

Maybe `AllowedVersions` is enough for him, but I don't know that yet.

Reply:
You are right, I will re-think this part.